[Bug] The release build is broken on macOS

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.


### Version

Doris Version: master
Compiler:
```shell
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: arm64-apple-darwin22.3.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

### What's Wrong?

```shell
Undefined symbols for architecture arm64:
  "void doris::vectorized::Decoder::init_decimal_converter<int>(std::__1::shared_ptr<doris::vectorized::IDataType const>&)", referenced from:
      doris::Status doris::vectorized::ByteArrayDictDecoder::_decode_binary_decimal<int>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(byte_array_dict_decoder.cpp.o)
      doris::Status doris::vectorized::ByteArrayPlainDecoder::_decode_binary_decimal<int>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(byte_array_plain_decoder.cpp.o)
      doris::Status doris::vectorized::FixLengthPlainDecoder::_decode_binary_decimal<int>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(fix_length_plain_decoder.cpp.o)
      doris::Status doris::vectorized::FixLengthPlainDecoder::_decode_primitive_decimal<int, int>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(fix_length_plain_decoder.cpp.o)
      doris::Status doris::vectorized::FixLengthPlainDecoder::_decode_primitive_decimal<int, long long>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(fix_length_plain_decoder.cpp.o)
  "void doris::vectorized::Decoder::init_decimal_converter<__int128>(std::__1::shared_ptr<doris::vectorized::IDataType const>&)", referenced from:
      doris::Status doris::vectorized::ByteArrayDictDecoder::_decode_binary_decimal<__int128>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(byte_array_dict_decoder.cpp.o)
      doris::Status doris::vectorized::ByteArrayPlainDecoder::_decode_binary_decimal<__int128>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(byte_array_plain_decoder.cpp.o)
      doris::Status doris::vectorized::FixLengthPlainDecoder::_decode_binary_decimal<__int128>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(fix_length_plain_decoder.cpp.o)
      doris::Status doris::vectorized::FixLengthPlainDecoder::_decode_primitive_decimal<__int128, int>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(fix_length_plain_decoder.cpp.o)
      doris::Status doris::vectorized::FixLengthPlainDecoder::_decode_primitive_decimal<__int128, long long>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(fix_length_plain_decoder.cpp.o)
  "void doris::vectorized::Decoder::init_decimal_converter<long long>(std::__1::shared_ptr<doris::vectorized::IDataType const>&)", referenced from:
      doris::Status doris::vectorized::ByteArrayDictDecoder::_decode_binary_decimal<long long>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(byte_array_dict_decoder.cpp.o)
      doris::Status doris::vectorized::ByteArrayPlainDecoder::_decode_binary_decimal<long long>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(byte_array_plain_decoder.cpp.o)
      doris::Status doris::vectorized::FixLengthPlainDecoder::_decode_binary_decimal<long long>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(fix_length_plain_decoder.cpp.o)
      doris::Status doris::vectorized::FixLengthPlainDecoder::_decode_primitive_decimal<long long, int>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(fix_length_plain_decoder.cpp.o)
      doris::Status doris::vectorized::FixLengthPlainDecoder::_decode_primitive_decimal<long long, long long>(COW<doris::vectorized::IColumn>::mutable_ptr<doris::vectorized::IColumn>&, std::__1::shared_ptr<doris::vectorized::IDataType const>&, doris::vectorized::ColumnSelectVector&) in libVec.a(fix_length_plain_decoder.cpp.o)
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

### What You Expected?

Build BE (with RELEASE build type) successfully.

### How to Reproduce?

```shell
BUILD_TYPE=RELEASE ./build.sh --clean --be -j 6
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/be/src/vec/exec/format/parquet/decoder.cpp
+++ b/be/src/vec/exec/format/parquet/decoder.cpp
@@ -132,27 +132,4 @@ void Decoder::init(FieldSchema* field_schema, cctz::time_zone* ctz) {
         }
     }
 }
-template <typename DecimalPrimitiveType>
-void Decoder::init_decimal_converter(DataTypePtr& data_type) {
-    if (_decode_params == nullptr || _field_schema == nullptr ||
-        _decode_params->decimal_scale.scale_type != DecimalScaleParams::NOT_INIT) {
-        return;
-    }
-    auto scale = _field_schema->parquet_schema.scale;
-    auto* decimal_type = reinterpret_cast<DataTypeDecimal<Decimal<DecimalPrimitiveType>>*>(
-            const_cast<IDataType*>(remove_nullable(data_type).get()));
-    auto dest_scale = decimal_type->get_scale();
-    if (dest_scale > scale) {
-        _decode_params->decimal_scale.scale_type = DecimalScaleParams::SCALE_UP;
-        _decode_params->decimal_scale.scale_factor =
-                DecimalScaleParams::get_scale_factor<DecimalPrimitiveType>(dest_scale - scale);
-    } else if (dest_scale < scale) {
-        _decode_params->decimal_scale.scale_type = DecimalScaleParams::SCALE_DOWN;
-        _decode_params->decimal_scale.scale_factor =
-                DecimalScaleParams::get_scale_factor<DecimalPrimitiveType>(scale - dest_scale);
-    } else {
-        _decode_params->decimal_scale.scale_type = DecimalScaleParams::NO_SCALE;
-        _decode_params->decimal_scale.scale_factor = 1;
-    }
-}
 } // namespace doris::vectorized

--- a/be/src/vec/exec/format/parquet/decoder.h
+++ b/be/src/vec/exec/format/parquet/decoder.h
@@ -92,6 +92,30 @@ protected:
     std::unique_ptr<DecodeParams> _decode_params = nullptr;
 };
 
+template <typename DecimalPrimitiveType>
+void Decoder::init_decimal_converter(DataTypePtr& data_type) {
+    if (_decode_params == nullptr || _field_schema == nullptr ||
+        _decode_params->decimal_scale.scale_type != DecimalScaleParams::NOT_INIT) {
+        return;
+    }
+    auto scale = _field_schema->parquet_schema.scale;
+    auto* decimal_type = reinterpret_cast<DataTypeDecimal<Decimal<DecimalPrimitiveType>>*>(
+            const_cast<IDataType*>(remove_nullable(data_type).get()));
+    auto dest_scale = decimal_type->get_scale();
+    if (dest_scale > scale) {
+        _decode_params->decimal_scale.scale_type = DecimalScaleParams::SCALE_UP;
+        _decode_params->decimal_scale.scale_factor =
+                DecimalScaleParams::get_scale_factor<DecimalPrimitiveType>(dest_scale - scale);
+    } else if (dest_scale < scale) {
+        _decode_params->decimal_scale.scale_type = DecimalScaleParams::SCALE_DOWN;
+        _decode_params->decimal_scale.scale_factor =
+                DecimalScaleParams::get_scale_factor<DecimalPrimitiveType>(scale - dest_scale);
+    } else {
+        _decode_params->decimal_scale.scale_type = DecimalScaleParams::NO_SCALE;
+        _decode_params->decimal_scale.scale_factor = 1;
+    }
+}
+
 class BaseDictDecoder : public Decoder {
 public:
     BaseDictDecoder() = default;

